### PR TITLE
Remove docker from bindep.txt

### PR DESCRIPTION
**Pull request by ssbarnea**
Putting docker inside bindep can conflict with other docker installations like docker-ce, which is now installed by default by ansible-tox-molecule role.

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/304

### CODE REVIEW

**Comment by ssbarnea** [comment]
@pabelanger ^^^

File: bindep.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # This is a cross-platform list tracking distribution packages needed by tests;
 # see https://docs.openstack.org/infra/bindep/ for additional information.
 
-docker [test platform:fedora]
+# do not put docker in default or test labels because it could conflict
```
Comment (pabelanger) [inline bindep.txt]: lets just delete it, nothing else outside of zuul is using bindep.txt

File: tools/test-setup.sh [hunk 2]
```diff
@@ -18,4 +18,6 @@ if ! [ $(getent group docker) ]; then
     sudo groupadd docker
 fi
 sudo gpasswd -a ${USER} docker
-sudo service docker restart
+# We do not fail here because ansible-tox-molecule role installs docker-ce
```
Comment (pabelanger) [inline tools/test-setup.sh]: same, this file can now be deleted, as install-docker roles sets up everything

File: bindep.txt [hunk 3]
```diff
@@ -1,6 +0,0 @@
-# This is a cross-platform list tracking distribution packages needed by tests;
-# see https://docs.openstack.org/infra/bindep/ for additional information.
-
-docker [test platform:fedora]
```
Comment (pabelanger) [inline bindep.txt:4]: sorry, we should only remove 'docker' everything else is still used.

**Pull request merged**